src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-            if (withdrawalAmountGwei >= REQUIRED_BALANCE_GWEI) {
-                // then the excess is immediately withdrawable
-                amountToSend = uint256(withdrawalAmountGwei - REQUIRED_BALANCE_GWEI) * uint256(GWEI_TO_WEI);
-                // and the extra execution layer ETH in the contract is REQUIRED_BALANCE_GWEI, which must be withdrawn through EigenLayer's normal withdrawal process
-                restakedExecutionLayerGwei += REQUIRED_BALANCE_GWEI;
-                /**
-                 * since in `verifyOvercommittedStake` the podOwner's beaconChainETH shares are decremented by `REQUIRED_BALANCE_WEI`, we must reverse the process here,
-                 * in order to allow the podOwner to complete their withdrawal through EigenLayer's normal withdrawal process
-                 */
-                eigenPodManager.restakeBeaconChainETH(podOwner, REQUIRED_BALANCE_WEI);
-            } else {
-                // otherwise, just use the full withdrawal amount to continue to "back" the podOwner's remaining shares in EigenLayer (i.e. none is instantly withdrawable)
-                restakedExecutionLayerGwei += withdrawalAmountGwei;
-                /**
-                 * since in `verifyOvercommittedStake` the podOwner's beaconChainETH shares are decremented by `REQUIRED_BALANCE_WEI`, we must reverse the process here,
-                 * in order to allow the podOwner to complete their withdrawal through EigenLayer's normal withdrawal process
-                 */
-                eigenPodManager.restakeBeaconChainETH(podOwner, uint256(withdrawalAmountGwei) * GWEI_TO_WEI);
+            // if the amount being withdrawn is not equal to the current accounted for validator balance, an update must be made
+            if (currentValidatorRestakedBalanceWei != withdrawalAmountWei) {
+                int256 sharesDelta = _calculateSharesDelta(withdrawalAmountWei, currentValidatorRestakedBalanceWei);
+                //update podOwner's shares in the strategy manager
+                eigenPodManager.recordBeaconChainETHBalanceUpdate(podOwner, beaconChainETHStrategyIndex, sharesDelta);
             }
+
         // If the validator status is withdrawn, they have already processed their ETH withdrawal
         }  else {
-            revert("EigenPod.verifyBeaconChainFullWithdrawal: VALIDATOR_STATUS is WITHDRAWN or invalid VALIDATOR_STATUS");
+            revert("EigenPod.verifyBeaconChainFullWithdrawal: VALIDATOR_STATUS is invalid VALIDATOR_STATUS");
         }
-
         // set the ETH validator status to withdrawn
-        validatorStatus[validatorIndex] = VALIDATOR_STATUS.WITHDRAWN;
+        _validatorPubkeyHashToInfo[validatorPubkeyHash].status = VALIDATOR_STATUS.WITHDRAWN;
+        // now that the validator has been proven to be withdrawn, we can set their restaked balance to 0
+        _validatorPubkeyHashToInfo[validatorPubkeyHash].restakedBalanceGwei = 0;
```

Comment:
I need you to convince me that this makes sense. aren't they still credited with shares in EigenLayer? maybe I'm just not understanding the purpose of this variable well enough, but that's not a great sign

Reply:
This variable tracks what's not yet queued for withdrawal.  So we set it to zero and increment withdrawableRestakedExecutionLayerGwei.  Secondly, in the edge case where you deposit more money into a validator after a full withdrawal, not setting this to zero fucks up the accounting because then sharesDelta is not computed correctly.  If _validatorPubkeyHashToInfo[validatorPubkeyHash].restakedBalanceGwei is set to 0, then sharesDelta = withdrawalAmountWei, and the withdrawal can be queued correctly.

Reply:
OK, seems correct, will can still definitely clarify documentation